From 440169d681cc9f422e8486f1324079bef0e0c35a Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Wed, 25 Feb 2015 00:01:15 +0000
Subject: [PATCH 3/3] Fix ldbl-128ibm acoshl inaccuracy (bug 18019).

The ldbl-128ibm implementation of acoshl uses a cut-off of 0x1p28 to
determine when to use log(x) + log(2) as a formula.  That cut-off is
too small for this format, resulting in large errors.  This patch
changes it to a more appropriate cut-off of 0x1p56, adding tests
around the cut-offs for various floating-point formats.

Tested for powerpc.  Also tested for x86_64 and x86 and updated ulps.

	[BZ #18019]
	* sysdeps/ieee754/ldbl-128ibm/e_acoshl.c (__ieee754_acoshl): Use
	2**56 not 2**28 as threshold for log (2x) formula.
	* math/auto-libm-test-in: Add more tests of acosh.
	* math/auto-libm-test-out: Regenerated.
	* sysdeps/i386/fpu/libm-test-ulps: Update.
	* sysdeps/x86_64/fpu/libm-test-ulps: Likewise.
---
 math/auto-libm-test-in                 |  31 ++
 math/auto-libm-test-out                | 835 +++++++++++++++++++++++++++++++++
 sysdeps/i386/fpu/libm-test-ulps        |   4 +
 sysdeps/ieee754/ldbl-128ibm/e_acoshl.c |   4 +-
 sysdeps/x86_64/fpu/libm-test-ulps      |  17 +-
 7 files changed, 899 insertions(+), 5 deletions(-)

diff --git a/math/auto-libm-test-in b/math/auto-libm-test-in
index 62e1439..a4bd972 100644
--- a/math/auto-libm-test-in
+++ b/math/auto-libm-test-in
@@ -43,6 +43,37 @@ acosh 1.625
 acosh 7
 acosh 100
 acosh 1e5
+acosh 0x1p8
+acosh 0x1p9
+acosh 0x1p10
+acosh 0x1p11
+acosh 0x1p12
+acosh 0x1p13
+acosh 0x1p24
+acosh 0x1p25
+acosh 0x1p26
+acosh 0x1p27
+acosh 0x1p28
+acosh 0x1p29
+acosh 0x1p30
+acosh 0x1p31
+acosh 0x1p32
+acosh 0x1p33
+acosh 0x1p48
+acosh 0x1p49
+acosh 0x1p50
+acosh 0x1p51
+acosh 0x1p52
+acosh 0x1p53
+acosh 0x1p54
+acosh 0x1p55
+acosh 0x1p56
+acosh 0x1p57
+acosh 0x1p58
+acosh 0x1p59
+acosh 0x1p100
+acosh 0x1p500
+acosh 0x1p5000
 acosh max no-test-inline
 
 asin 0
diff --git a/math/auto-libm-test-out b/math/auto-libm-test-out
index 9d00c3b..c8bc6a8 100644
--- a/math/auto-libm-test-out
+++ b/math/auto-libm-test-out
@@ -1280,6 +1280,841 @@ acosh 1e5
 = acosh tonearest ldbl-128ibm 0x1.86ap+16L : 0xc.34c12d490b800b9c3c81561408p+0L : inexact-ok
 = acosh towardzero ldbl-128ibm 0x1.86ap+16L : 0xc.34c12d490b800b9c3c81561408p+0L : inexact-ok
 = acosh upward ldbl-128ibm 0x1.86ap+16L : 0xc.34c12d490b800b9c3c8156140cp+0L : inexact-ok
+acosh 0x1p8
+= acosh downward flt-32 0x1p+8f : 0x6.3d029p+0f : inexact-ok
+= acosh tonearest flt-32 0x1p+8f : 0x6.3d0298p+0f : inexact-ok
+= acosh towardzero flt-32 0x1p+8f : 0x6.3d029p+0f : inexact-ok
+= acosh upward flt-32 0x1p+8f : 0x6.3d0298p+0f : inexact-ok
+= acosh downward dbl-64 0x1p+8 : 0x6.3d0297b6484bp+0 : inexact-ok
+= acosh tonearest dbl-64 0x1p+8 : 0x6.3d0297b6484b4p+0 : inexact-ok
+= acosh towardzero dbl-64 0x1p+8 : 0x6.3d0297b6484bp+0 : inexact-ok
+= acosh upward dbl-64 0x1p+8 : 0x6.3d0297b6484b4p+0 : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+8L : 0x6.3d0297b6484b39bp+0L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+8L : 0x6.3d0297b6484b39b8p+0L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+8L : 0x6.3d0297b6484b39bp+0L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+8L : 0x6.3d0297b6484b39b8p+0L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+8L : 0x6.3d0297b6484b39bp+0L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+8L : 0x6.3d0297b6484b39b8p+0L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+8L : 0x6.3d0297b6484b39bp+0L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+8L : 0x6.3d0297b6484b39b8p+0L : inexact-ok
+= acosh downward ldbl-128 0x1p+8L : 0x6.3d0297b6484b39b4baebf4b5fc98p+0L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+8L : 0x6.3d0297b6484b39b4baebf4b5fc98p+0L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+8L : 0x6.3d0297b6484b39b4baebf4b5fc98p+0L : inexact-ok
+= acosh upward ldbl-128 0x1p+8L : 0x6.3d0297b6484b39b4baebf4b5fc9cp+0L : inexact-ok
+= acosh downward ldbl-128ibm 0x1p+8L : 0x6.3d0297b6484b39b4baebf4b5fcp+0L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x1p+8L : 0x6.3d0297b6484b39b4baebf4b5fcp+0L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x1p+8L : 0x6.3d0297b6484b39b4baebf4b5fcp+0L : inexact-ok
+= acosh upward ldbl-128ibm 0x1p+8L : 0x6.3d0297b6484b39b4baebf4b5fep+0L : inexact-ok
+acosh 0x1p9
+= acosh downward flt-32 0x2p+8f : 0x6.ee74d8p+0f : inexact-ok
+= acosh tonearest flt-32 0x2p+8f : 0x6.ee74ep+0f : inexact-ok
+= acosh towardzero flt-32 0x2p+8f : 0x6.ee74d8p+0f : inexact-ok
+= acosh upward flt-32 0x2p+8f : 0x6.ee74ep+0f : inexact-ok
+= acosh downward dbl-64 0x2p+8 : 0x6.ee74dfae309acp+0 : inexact-ok
+= acosh tonearest dbl-64 0x2p+8 : 0x6.ee74dfae309acp+0 : inexact-ok
+= acosh towardzero dbl-64 0x2p+8 : 0x6.ee74dfae309acp+0 : inexact-ok
+= acosh upward dbl-64 0x2p+8 : 0x6.ee74dfae309bp+0 : inexact-ok
+= acosh downward ldbl-96-intel 0x2p+8L : 0x6.ee74dfae309ac08p+0L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x2p+8L : 0x6.ee74dfae309ac08p+0L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x2p+8L : 0x6.ee74dfae309ac08p+0L : inexact-ok
+= acosh upward ldbl-96-intel 0x2p+8L : 0x6.ee74dfae309ac088p+0L : inexact-ok
+= acosh downward ldbl-96-m68k 0x2p+8L : 0x6.ee74dfae309ac08p+0L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x2p+8L : 0x6.ee74dfae309ac08p+0L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x2p+8L : 0x6.ee74dfae309ac08p+0L : inexact-ok
+= acosh upward ldbl-96-m68k 0x2p+8L : 0x6.ee74dfae309ac088p+0L : inexact-ok
+= acosh downward ldbl-128 0x2p+8L : 0x6.ee74dfae309ac0808d86ee993ef4p+0L : inexact-ok
+= acosh tonearest ldbl-128 0x2p+8L : 0x6.ee74dfae309ac0808d86ee993ef4p+0L : inexact-ok
+= acosh towardzero ldbl-128 0x2p+8L : 0x6.ee74dfae309ac0808d86ee993ef4p+0L : inexact-ok
+= acosh upward ldbl-128 0x2p+8L : 0x6.ee74dfae309ac0808d86ee993ef8p+0L : inexact-ok
+= acosh downward ldbl-128ibm 0x2p+8L : 0x6.ee74dfae309ac0808d86ee993ep+0L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x2p+8L : 0x6.ee74dfae309ac0808d86ee993ep+0L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x2p+8L : 0x6.ee74dfae309ac0808d86ee993ep+0L : inexact-ok
+= acosh upward ldbl-128ibm 0x2p+8L : 0x6.ee74dfae309ac0808d86ee994p+0L : inexact-ok
+acosh 0x1p10
+= acosh downward flt-32 0x4p+8f : 0x7.9fe7p+0f : inexact-ok
+= acosh tonearest flt-32 0x4p+8f : 0x7.9fe7p+0f : inexact-ok
+= acosh towardzero flt-32 0x4p+8f : 0x7.9fe7p+0f : inexact-ok
+= acosh upward flt-32 0x4p+8f : 0x7.9fe708p+0f : inexact-ok
+= acosh downward dbl-64 0x4p+8 : 0x7.9fe703a603d2p+0 : inexact-ok
+= acosh tonearest dbl-64 0x4p+8 : 0x7.9fe703a603d24p+0 : inexact-ok
+= acosh towardzero dbl-64 0x4p+8 : 0x7.9fe703a603d2p+0 : inexact-ok
+= acosh upward dbl-64 0x4p+8 : 0x7.9fe703a603d24p+0 : inexact-ok
+= acosh downward ldbl-96-intel 0x4p+8L : 0x7.9fe703a603d23a6p+0L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x4p+8L : 0x7.9fe703a603d23a6p+0L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x4p+8L : 0x7.9fe703a603d23a6p+0L : inexact-ok
+= acosh upward ldbl-96-intel 0x4p+8L : 0x7.9fe703a603d23a68p+0L : inexact-ok
+= acosh downward ldbl-96-m68k 0x4p+8L : 0x7.9fe703a603d23a6p+0L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x4p+8L : 0x7.9fe703a603d23a6p+0L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x4p+8L : 0x7.9fe703a603d23a6p+0L : inexact-ok
+= acosh upward ldbl-96-m68k 0x4p+8L : 0x7.9fe703a603d23a68p+0L : inexact-ok
+= acosh downward ldbl-128 0x4p+8L : 0x7.9fe703a603d23a60d7735972d5b4p+0L : inexact-ok
+= acosh tonearest ldbl-128 0x4p+8L : 0x7.9fe703a603d23a60d7735972d5b8p+0L : inexact-ok
+= acosh towardzero ldbl-128 0x4p+8L : 0x7.9fe703a603d23a60d7735972d5b4p+0L : inexact-ok
+= acosh upward ldbl-128 0x4p+8L : 0x7.9fe703a603d23a60d7735972d5b8p+0L : inexact-ok
+= acosh downward ldbl-128ibm 0x4p+8L : 0x7.9fe703a603d23a60d7735972d4p+0L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x4p+8L : 0x7.9fe703a603d23a60d7735972d6p+0L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x4p+8L : 0x7.9fe703a603d23a60d7735972d4p+0L : inexact-ok
+= acosh upward ldbl-128ibm 0x4p+8L : 0x7.9fe703a603d23a60d7735972d6p+0L : inexact-ok
+acosh 0x1p11
+= acosh downward flt-32 0x8p+8f : 0x8.51591p+0f : inexact-ok
+= acosh tonearest flt-32 0x8p+8f : 0x8.51592p+0f : inexact-ok
+= acosh towardzero flt-32 0x8p+8f : 0x8.51591p+0f : inexact-ok
+= acosh upward flt-32 0x8p+8f : 0x8.51592p+0f : inexact-ok
+= acosh downward dbl-64 0x8p+8 : 0x8.51591e9dd5b8p+0 : inexact-ok
+= acosh tonearest dbl-64 0x8p+8 : 0x8.51591e9dd5b8p+0 : inexact-ok
+= acosh towardzero dbl-64 0x8p+8 : 0x8.51591e9dd5b8p+0 : inexact-ok
+= acosh upward dbl-64 0x8p+8 : 0x8.51591e9dd5b88p+0 : inexact-ok
+= acosh downward ldbl-96-intel 0x8p+8L : 0x8.51591e9dd5b834p+0L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x8p+8L : 0x8.51591e9dd5b8341p+0L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x8p+8L : 0x8.51591e9dd5b834p+0L : inexact-ok
+= acosh upward ldbl-96-intel 0x8p+8L : 0x8.51591e9dd5b8341p+0L : inexact-ok
+= acosh downward ldbl-96-m68k 0x8p+8L : 0x8.51591e9dd5b834p+0L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x8p+8L : 0x8.51591e9dd5b8341p+0L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x8p+8L : 0x8.51591e9dd5b834p+0L : inexact-ok
+= acosh upward ldbl-96-m68k 0x8p+8L : 0x8.51591e9dd5b8341p+0L : inexact-ok
+= acosh downward ldbl-128 0x8p+8L : 0x8.51591e9dd5b8340d735715c21a08p+0L : inexact-ok
+= acosh tonearest ldbl-128 0x8p+8L : 0x8.51591e9dd5b8340d735715c21a1p+0L : inexact-ok
+= acosh towardzero ldbl-128 0x8p+8L : 0x8.51591e9dd5b8340d735715c21a08p+0L : inexact-ok
+= acosh upward ldbl-128 0x8p+8L : 0x8.51591e9dd5b8340d735715c21a1p+0L : inexact-ok
+= acosh downward ldbl-128ibm 0x8p+8L : 0x8.51591e9dd5b8340d735715c218p+0L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x8p+8L : 0x8.51591e9dd5b8340d735715c21cp+0L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x8p+8L : 0x8.51591e9dd5b8340d735715c218p+0L : inexact-ok
+= acosh upward ldbl-128ibm 0x8p+8L : 0x8.51591e9dd5b8340d735715c21cp+0L : inexact-ok
+acosh 0x1p12
+= acosh downward flt-32 0x1p+12f : 0x9.02cb3p+0f : inexact-ok
+= acosh tonearest flt-32 0x1p+12f : 0x9.02cb3p+0f : inexact-ok
+= acosh towardzero flt-32 0x1p+12f : 0x9.02cb3p+0f : inexact-ok
+= acosh upward flt-32 0x1p+12f : 0x9.02cb4p+0f : inexact-ok
+= acosh downward dbl-64 0x1p+12 : 0x9.02cb3755a789p+0 : inexact-ok
+= acosh tonearest dbl-64 0x1p+12 : 0x9.02cb3755a789p+0 : inexact-ok
+= acosh towardzero dbl-64 0x1p+12 : 0x9.02cb3755a789p+0 : inexact-ok
+= acosh upward dbl-64 0x1p+12 : 0x9.02cb3755a7898p+0 : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+12L : 0x9.02cb3755a78915bp+0L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+12L : 0x9.02cb3755a78915cp+0L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+12L : 0x9.02cb3755a78915bp+0L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+12L : 0x9.02cb3755a78915cp+0L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+12L : 0x9.02cb3755a78915bp+0L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+12L : 0x9.02cb3755a78915cp+0L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+12L : 0x9.02cb3755a78915bp+0L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+12L : 0x9.02cb3755a78915cp+0L : inexact-ok
+= acosh downward ldbl-128 0x1p+12L : 0x9.02cb3755a78915b94082c962d54p+0L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+12L : 0x9.02cb3755a78915b94082c962d54p+0L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+12L : 0x9.02cb3755a78915b94082c962d54p+0L : inexact-ok
+= acosh upward ldbl-128 0x1p+12L : 0x9.02cb3755a78915b94082c962d548p+0L : inexact-ok
+= acosh downward ldbl-128ibm 0x1p+12L : 0x9.02cb3755a78915b94082c962d4p+0L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x1p+12L : 0x9.02cb3755a78915b94082c962d4p+0L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x1p+12L : 0x9.02cb3755a78915b94082c962d4p+0L : inexact-ok
+= acosh upward ldbl-128ibm 0x1p+12L : 0x9.02cb3755a78915b94082c962d8p+0L : inexact-ok
+acosh 0x1p13
+= acosh downward flt-32 0x2p+12f : 0x9.b43d4p+0f : inexact-ok
+= acosh tonearest flt-32 0x2p+12f : 0x9.b43d5p+0f : inexact-ok
+= acosh towardzero flt-32 0x2p+12f : 0x9.b43d4p+0f : inexact-ok
+= acosh upward flt-32 0x2p+12f : 0x9.b43d5p+0f : inexact-ok
+= acosh downward dbl-64 0x2p+12 : 0x9.b43d4f7d79588p+0 : inexact-ok
+= acosh tonearest dbl-64 0x2p+12 : 0x9.b43d4f7d79588p+0 : inexact-ok
+= acosh towardzero dbl-64 0x2p+12 : 0x9.b43d4f7d79588p+0 : inexact-ok
+= acosh upward dbl-64 0x2p+12 : 0x9.b43d4f7d7959p+0 : inexact-ok
+= acosh downward ldbl-96-intel 0x2p+12L : 0x9.b43d4f7d7958a5ep+0L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x2p+12L : 0x9.b43d4f7d7958a5ep+0L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x2p+12L : 0x9.b43d4f7d7958a5ep+0L : inexact-ok
+= acosh upward ldbl-96-intel 0x2p+12L : 0x9.b43d4f7d7958a5fp+0L : inexact-ok
+= acosh downward ldbl-96-m68k 0x2p+12L : 0x9.b43d4f7d7958a5ep+0L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x2p+12L : 0x9.b43d4f7d7958a5ep+0L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x2p+12L : 0x9.b43d4f7d7958a5ep+0L : inexact-ok
+= acosh upward ldbl-96-m68k 0x2p+12L : 0x9.b43d4f7d7958a5fp+0L : inexact-ok
+= acosh downward ldbl-128 0x2p+12L : 0x9.b43d4f7d7958a5e50a739cfae1e8p+0L : inexact-ok
+= acosh tonearest ldbl-128 0x2p+12L : 0x9.b43d4f7d7958a5e50a739cfae1e8p+0L : inexact-ok
+= acosh towardzero ldbl-128 0x2p+12L : 0x9.b43d4f7d7958a5e50a739cfae1e8p+0L : inexact-ok
+= acosh upward ldbl-128 0x2p+12L : 0x9.b43d4f7d7958a5e50a739cfae1fp+0L : inexact-ok
+= acosh downward ldbl-128ibm 0x2p+12L : 0x9.b43d4f7d7958a5e50a739cfaep+0L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x2p+12L : 0x9.b43d4f7d7958a5e50a739cfaep+0L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x2p+12L : 0x9.b43d4f7d7958a5e50a739cfaep+0L : inexact-ok
+= acosh upward ldbl-128ibm 0x2p+12L : 0x9.b43d4f7d7958a5e50a739cfae4p+0L : inexact-ok
+acosh 0x1p24
+= acosh downward flt-32 0x1p+24f : 0x1.154244p+4f : inexact-ok
+= acosh tonearest flt-32 0x1p+24f : 0x1.154246p+4f : inexact-ok
+= acosh towardzero flt-32 0x1p+24f : 0x1.154244p+4f : inexact-ok
+= acosh upward flt-32 0x1p+24f : 0x1.154246p+4f : inexact-ok
+= acosh downward dbl-64 0x1p+24 : 0x1.1542457337d42p+4 : inexact-ok
+= acosh tonearest dbl-64 0x1p+24 : 0x1.1542457337d43p+4 : inexact-ok
+= acosh towardzero dbl-64 0x1p+24 : 0x1.1542457337d42p+4 : inexact-ok
+= acosh upward dbl-64 0x1p+24 : 0x1.1542457337d43p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+24L : 0x1.1542457337d42a1cp+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+24L : 0x1.1542457337d42a1cp+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+24L : 0x1.1542457337d42a1cp+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+24L : 0x1.1542457337d42a1ep+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+24L : 0x1.1542457337d42a1cp+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+24L : 0x1.1542457337d42a1cp+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+24L : 0x1.1542457337d42a1cp+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+24L : 0x1.1542457337d42a1ep+4L : inexact-ok
+= acosh downward ldbl-128 0x1p+24L : 0x1.1542457337d42a1c6b73c89d84abp+4L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+24L : 0x1.1542457337d42a1c6b73c89d84acp+4L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+24L : 0x1.1542457337d42a1c6b73c89d84abp+4L : inexact-ok
+= acosh upward ldbl-128 0x1p+24L : 0x1.1542457337d42a1c6b73c89d84acp+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x1p+24L : 0x1.1542457337d42a1c6b73c89d848p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x1p+24L : 0x1.1542457337d42a1c6b73c89d848p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x1p+24L : 0x1.1542457337d42a1c6b73c89d848p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x1p+24L : 0x1.1542457337d42a1c6b73c89d85p+4L : inexact-ok
+acosh 0x1p25
+= acosh downward flt-32 0x2p+24f : 0x1.205966p+4f : inexact-ok
+= acosh tonearest flt-32 0x2p+24f : 0x1.205966p+4f : inexact-ok
+= acosh towardzero flt-32 0x2p+24f : 0x1.205966p+4f : inexact-ok
+= acosh upward flt-32 0x2p+24f : 0x1.205968p+4f : inexact-ok
+= acosh downward dbl-64 0x2p+24 : 0x1.205966f2b4f12p+4 : inexact-ok
+= acosh tonearest dbl-64 0x2p+24 : 0x1.205966f2b4f12p+4 : inexact-ok
+= acosh towardzero dbl-64 0x2p+24 : 0x1.205966f2b4f12p+4 : inexact-ok
+= acosh upward dbl-64 0x2p+24 : 0x1.205966f2b4f13p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x2p+24L : 0x1.205966f2b4f124b6p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x2p+24L : 0x1.205966f2b4f124b8p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x2p+24L : 0x1.205966f2b4f124b6p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x2p+24L : 0x1.205966f2b4f124b8p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x2p+24L : 0x1.205966f2b4f124b6p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x2p+24L : 0x1.205966f2b4f124b8p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x2p+24L : 0x1.205966f2b4f124b6p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x2p+24L : 0x1.205966f2b4f124b8p+4L : inexact-ok
+= acosh downward ldbl-128 0x2p+24L : 0x1.205966f2b4f124b7281203d70652p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x2p+24L : 0x1.205966f2b4f124b7281203d70653p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x2p+24L : 0x1.205966f2b4f124b7281203d70652p+4L : inexact-ok
+= acosh upward ldbl-128 0x2p+24L : 0x1.205966f2b4f124b7281203d70653p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x2p+24L : 0x1.205966f2b4f124b7281203d706p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x2p+24L : 0x1.205966f2b4f124b7281203d7068p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x2p+24L : 0x1.205966f2b4f124b7281203d706p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x2p+24L : 0x1.205966f2b4f124b7281203d7068p+4L : inexact-ok
+acosh 0x1p26
+= acosh downward flt-32 0x4p+24f : 0x1.2b7088p+4f : inexact-ok
+= acosh tonearest flt-32 0x4p+24f : 0x1.2b7088p+4f : inexact-ok
+= acosh towardzero flt-32 0x4p+24f : 0x1.2b7088p+4f : inexact-ok
+= acosh upward flt-32 0x4p+24f : 0x1.2b708ap+4f : inexact-ok
+= acosh downward dbl-64 0x4p+24 : 0x1.2b708872320e1p+4 : inexact-ok
+= acosh tonearest dbl-64 0x4p+24 : 0x1.2b708872320e2p+4 : inexact-ok
+= acosh towardzero dbl-64 0x4p+24 : 0x1.2b708872320e1p+4 : inexact-ok
+= acosh upward dbl-64 0x4p+24 : 0x1.2b708872320e2p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x4p+24L : 0x1.2b708872320e1d1p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x4p+24L : 0x1.2b708872320e1d12p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x4p+24L : 0x1.2b708872320e1d1p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x4p+24L : 0x1.2b708872320e1d12p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x4p+24L : 0x1.2b708872320e1d1p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x4p+24L : 0x1.2b708872320e1d12p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x4p+24L : 0x1.2b708872320e1d1p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x4p+24L : 0x1.2b708872320e1d12p+4L : inexact-ok
+= acosh downward ldbl-128 0x4p+24L : 0x1.2b708872320e1d11e4b03f1086a8p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x4p+24L : 0x1.2b708872320e1d11e4b03f1086a9p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x4p+24L : 0x1.2b708872320e1d11e4b03f1086a8p+4L : inexact-ok
+= acosh upward ldbl-128 0x4p+24L : 0x1.2b708872320e1d11e4b03f1086a9p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x4p+24L : 0x1.2b708872320e1d11e4b03f10868p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x4p+24L : 0x1.2b708872320e1d11e4b03f10868p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x4p+24L : 0x1.2b708872320e1d11e4b03f10868p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x4p+24L : 0x1.2b708872320e1d11e4b03f1087p+4L : inexact-ok
+acosh 0x1p27
+= acosh downward flt-32 0x8p+24f : 0x1.3687a8p+4f : inexact-ok
+= acosh tonearest flt-32 0x8p+24f : 0x1.3687aap+4f : inexact-ok
+= acosh towardzero flt-32 0x8p+24f : 0x1.3687a8p+4f : inexact-ok
+= acosh upward flt-32 0x8p+24f : 0x1.3687aap+4f : inexact-ok
+= acosh downward dbl-64 0x8p+24 : 0x1.3687a9f1af2b1p+4 : inexact-ok
+= acosh tonearest dbl-64 0x8p+24 : 0x1.3687a9f1af2b1p+4 : inexact-ok
+= acosh towardzero dbl-64 0x8p+24 : 0x1.3687a9f1af2b1p+4 : inexact-ok
+= acosh upward dbl-64 0x8p+24 : 0x1.3687a9f1af2b2p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x8p+24L : 0x1.3687a9f1af2b14dcp+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x8p+24L : 0x1.3687a9f1af2b14dcp+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x8p+24L : 0x1.3687a9f1af2b14dcp+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x8p+24L : 0x1.3687a9f1af2b14dep+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x8p+24L : 0x1.3687a9f1af2b14dcp+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x8p+24L : 0x1.3687a9f1af2b14dcp+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x8p+24L : 0x1.3687a9f1af2b14dcp+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x8p+24L : 0x1.3687a9f1af2b14dep+4L : inexact-ok
+= acosh downward ldbl-128 0x8p+24L : 0x1.3687a9f1af2b14dca14e7a4a06e9p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x8p+24L : 0x1.3687a9f1af2b14dca14e7a4a06e9p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x8p+24L : 0x1.3687a9f1af2b14dca14e7a4a06e9p+4L : inexact-ok
+= acosh upward ldbl-128 0x8p+24L : 0x1.3687a9f1af2b14dca14e7a4a06eap+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x8p+24L : 0x1.3687a9f1af2b14dca14e7a4a068p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x8p+24L : 0x1.3687a9f1af2b14dca14e7a4a07p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x8p+24L : 0x1.3687a9f1af2b14dca14e7a4a068p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x8p+24L : 0x1.3687a9f1af2b14dca14e7a4a07p+4L : inexact-ok
+acosh 0x1p28
+= acosh downward flt-32 0x1p+28f : 0x1.419ecap+4f : inexact-ok
+= acosh tonearest flt-32 0x1p+28f : 0x1.419eccp+4f : inexact-ok
+= acosh towardzero flt-32 0x1p+28f : 0x1.419ecap+4f : inexact-ok
+= acosh upward flt-32 0x1p+28f : 0x1.419eccp+4f : inexact-ok
+= acosh downward dbl-64 0x1p+28 : 0x1.419ecb712c48p+4 : inexact-ok
+= acosh tonearest dbl-64 0x1p+28 : 0x1.419ecb712c481p+4 : inexact-ok
+= acosh towardzero dbl-64 0x1p+28 : 0x1.419ecb712c48p+4 : inexact-ok
+= acosh upward dbl-64 0x1p+28 : 0x1.419ecb712c481p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+28L : 0x1.419ecb712c480c82p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+28L : 0x1.419ecb712c480c84p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+28L : 0x1.419ecb712c480c82p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+28L : 0x1.419ecb712c480c84p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+28L : 0x1.419ecb712c480c82p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+28L : 0x1.419ecb712c480c84p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+28L : 0x1.419ecb712c480c82p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+28L : 0x1.419ecb712c480c84p+4L : inexact-ok
+= acosh downward ldbl-128 0x1p+28L : 0x1.419ecb712c480c835decb5838728p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+28L : 0x1.419ecb712c480c835decb5838728p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+28L : 0x1.419ecb712c480c835decb5838728p+4L : inexact-ok
+= acosh upward ldbl-128 0x1p+28L : 0x1.419ecb712c480c835decb5838729p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x1p+28L : 0x1.419ecb712c480c835decb58387p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x1p+28L : 0x1.419ecb712c480c835decb58387p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x1p+28L : 0x1.419ecb712c480c835decb58387p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x1p+28L : 0x1.419ecb712c480c835decb583878p+4L : inexact-ok
+acosh 0x1p29
+= acosh downward flt-32 0x2p+28f : 0x1.4cb5ecp+4f : inexact-ok
+= acosh tonearest flt-32 0x2p+28f : 0x1.4cb5ecp+4f : inexact-ok
+= acosh towardzero flt-32 0x2p+28f : 0x1.4cb5ecp+4f : inexact-ok
+= acosh upward flt-32 0x2p+28f : 0x1.4cb5eep+4f : inexact-ok
+= acosh downward dbl-64 0x2p+28 : 0x1.4cb5ecf0a965p+4 : inexact-ok
+= acosh tonearest dbl-64 0x2p+28 : 0x1.4cb5ecf0a965p+4 : inexact-ok
+= acosh towardzero dbl-64 0x2p+28 : 0x1.4cb5ecf0a965p+4 : inexact-ok
+= acosh upward dbl-64 0x2p+28 : 0x1.4cb5ecf0a9651p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x2p+28L : 0x1.4cb5ecf0a965042p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x2p+28L : 0x1.4cb5ecf0a9650422p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x2p+28L : 0x1.4cb5ecf0a965042p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x2p+28L : 0x1.4cb5ecf0a9650422p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x2p+28L : 0x1.4cb5ecf0a965042p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x2p+28L : 0x1.4cb5ecf0a9650422p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x2p+28L : 0x1.4cb5ecf0a965042p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x2p+28L : 0x1.4cb5ecf0a9650422p+4L : inexact-ok
+= acosh downward ldbl-128 0x2p+28L : 0x1.4cb5ecf0a96504211a8af0bd0767p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x2p+28L : 0x1.4cb5ecf0a96504211a8af0bd0768p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x2p+28L : 0x1.4cb5ecf0a96504211a8af0bd0767p+4L : inexact-ok
+= acosh upward ldbl-128 0x2p+28L : 0x1.4cb5ecf0a96504211a8af0bd0768p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x2p+28L : 0x1.4cb5ecf0a96504211a8af0bd07p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x2p+28L : 0x1.4cb5ecf0a96504211a8af0bd078p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x2p+28L : 0x1.4cb5ecf0a96504211a8af0bd07p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x2p+28L : 0x1.4cb5ecf0a96504211a8af0bd078p+4L : inexact-ok
+acosh 0x1p30
+= acosh downward flt-32 0x4p+28f : 0x1.57cd0ep+4f : inexact-ok
+= acosh tonearest flt-32 0x4p+28f : 0x1.57cd0ep+4f : inexact-ok
+= acosh towardzero flt-32 0x4p+28f : 0x1.57cd0ep+4f : inexact-ok
+= acosh upward flt-32 0x4p+28f : 0x1.57cd1p+4f : inexact-ok
+= acosh downward dbl-64 0x4p+28 : 0x1.57cd0e702681fp+4 : inexact-ok
+= acosh tonearest dbl-64 0x4p+28 : 0x1.57cd0e702682p+4 : inexact-ok
+= acosh towardzero dbl-64 0x4p+28 : 0x1.57cd0e702681fp+4 : inexact-ok
+= acosh upward dbl-64 0x4p+28 : 0x1.57cd0e702682p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x4p+28L : 0x1.57cd0e702681fbbcp+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x4p+28L : 0x1.57cd0e702681fbbcp+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x4p+28L : 0x1.57cd0e702681fbbcp+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x4p+28L : 0x1.57cd0e702681fbbep+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x4p+28L : 0x1.57cd0e702681fbbcp+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x4p+28L : 0x1.57cd0e702681fbbcp+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x4p+28L : 0x1.57cd0e702681fbbcp+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x4p+28L : 0x1.57cd0e702681fbbep+4L : inexact-ok
+= acosh downward ldbl-128 0x4p+28L : 0x1.57cd0e702681fbbc97292bf687a6p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x4p+28L : 0x1.57cd0e702681fbbc97292bf687a7p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x4p+28L : 0x1.57cd0e702681fbbc97292bf687a6p+4L : inexact-ok
+= acosh upward ldbl-128 0x4p+28L : 0x1.57cd0e702681fbbc97292bf687a7p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x4p+28L : 0x1.57cd0e702681fbbc97292bf6878p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x4p+28L : 0x1.57cd0e702681fbbc97292bf6878p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x4p+28L : 0x1.57cd0e702681fbbc97292bf6878p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x4p+28L : 0x1.57cd0e702681fbbc97292bf688p+4L : inexact-ok
+acosh 0x1p31
+= acosh downward flt-32 0x8p+28f : 0x1.62e42ep+4f : inexact-ok
+= acosh tonearest flt-32 0x8p+28f : 0x1.62e43p+4f : inexact-ok
+= acosh towardzero flt-32 0x8p+28f : 0x1.62e42ep+4f : inexact-ok
+= acosh upward flt-32 0x8p+28f : 0x1.62e43p+4f : inexact-ok
+= acosh downward dbl-64 0x8p+28 : 0x1.62e42fefa39efp+4 : inexact-ok
+= acosh tonearest dbl-64 0x8p+28 : 0x1.62e42fefa39efp+4 : inexact-ok
+= acosh towardzero dbl-64 0x8p+28 : 0x1.62e42fefa39efp+4 : inexact-ok
+= acosh upward dbl-64 0x8p+28 : 0x1.62e42fefa39fp+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x8p+28L : 0x1.62e42fefa39ef356p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x8p+28L : 0x1.62e42fefa39ef358p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x8p+28L : 0x1.62e42fefa39ef356p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x8p+28L : 0x1.62e42fefa39ef358p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x8p+28L : 0x1.62e42fefa39ef356p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x8p+28L : 0x1.62e42fefa39ef358p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x8p+28L : 0x1.62e42fefa39ef356p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x8p+28L : 0x1.62e42fefa39ef358p+4L : inexact-ok
+= acosh downward ldbl-128 0x8p+28L : 0x1.62e42fefa39ef35783c7673007e5p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x8p+28L : 0x1.62e42fefa39ef35783c7673007e6p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x8p+28L : 0x1.62e42fefa39ef35783c7673007e5p+4L : inexact-ok
+= acosh upward ldbl-128 0x8p+28L : 0x1.62e42fefa39ef35783c7673007e6p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x8p+28L : 0x1.62e42fefa39ef35783c76730078p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x8p+28L : 0x1.62e42fefa39ef35783c7673008p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x8p+28L : 0x1.62e42fefa39ef35783c76730078p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x8p+28L : 0x1.62e42fefa39ef35783c7673008p+4L : inexact-ok
+acosh 0x1p32
+= acosh downward flt-32 0x1p+32f : 0x1.6dfb5p+4f : inexact-ok
+= acosh tonearest flt-32 0x1p+32f : 0x1.6dfb52p+4f : inexact-ok
+= acosh towardzero flt-32 0x1p+32f : 0x1.6dfb5p+4f : inexact-ok
+= acosh upward flt-32 0x1p+32f : 0x1.6dfb52p+4f : inexact-ok
+= acosh downward dbl-64 0x1p+32 : 0x1.6dfb516f20bbep+4 : inexact-ok
+= acosh tonearest dbl-64 0x1p+32 : 0x1.6dfb516f20bbfp+4 : inexact-ok
+= acosh towardzero dbl-64 0x1p+32 : 0x1.6dfb516f20bbep+4 : inexact-ok
+= acosh upward dbl-64 0x1p+32 : 0x1.6dfb516f20bbfp+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+32L : 0x1.6dfb516f20bbeaf2p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+32L : 0x1.6dfb516f20bbeaf2p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+32L : 0x1.6dfb516f20bbeaf2p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+32L : 0x1.6dfb516f20bbeaf4p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+32L : 0x1.6dfb516f20bbeaf2p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+32L : 0x1.6dfb516f20bbeaf2p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+32L : 0x1.6dfb516f20bbeaf2p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+32L : 0x1.6dfb516f20bbeaf4p+4L : inexact-ok
+= acosh downward ldbl-128 0x1p+32L : 0x1.6dfb516f20bbeaf24c65a2698825p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+32L : 0x1.6dfb516f20bbeaf24c65a2698825p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+32L : 0x1.6dfb516f20bbeaf24c65a2698825p+4L : inexact-ok
+= acosh upward ldbl-128 0x1p+32L : 0x1.6dfb516f20bbeaf24c65a2698826p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x1p+32L : 0x1.6dfb516f20bbeaf24c65a26988p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x1p+32L : 0x1.6dfb516f20bbeaf24c65a26988p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x1p+32L : 0x1.6dfb516f20bbeaf24c65a26988p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x1p+32L : 0x1.6dfb516f20bbeaf24c65a269888p+4L : inexact-ok
+acosh 0x1p33
+= acosh downward flt-32 0x2p+32f : 0x1.791272p+4f : inexact-ok
+= acosh tonearest flt-32 0x2p+32f : 0x1.791272p+4f : inexact-ok
+= acosh towardzero flt-32 0x2p+32f : 0x1.791272p+4f : inexact-ok
+= acosh upward flt-32 0x2p+32f : 0x1.791274p+4f : inexact-ok
+= acosh downward dbl-64 0x2p+32 : 0x1.791272ee9dd8ep+4 : inexact-ok
+= acosh tonearest dbl-64 0x2p+32 : 0x1.791272ee9dd8ep+4 : inexact-ok
+= acosh towardzero dbl-64 0x2p+32 : 0x1.791272ee9dd8ep+4 : inexact-ok
+= acosh upward dbl-64 0x2p+32 : 0x1.791272ee9dd8fp+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x2p+32L : 0x1.791272ee9dd8e28cp+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x2p+32L : 0x1.791272ee9dd8e28ep+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x2p+32L : 0x1.791272ee9dd8e28cp+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x2p+32L : 0x1.791272ee9dd8e28ep+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x2p+32L : 0x1.791272ee9dd8e28cp+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x2p+32L : 0x1.791272ee9dd8e28ep+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x2p+32L : 0x1.791272ee9dd8e28cp+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x2p+32L : 0x1.791272ee9dd8e28ep+4L : inexact-ok
+= acosh downward ldbl-128 0x2p+32L : 0x1.791272ee9dd8e28d0c03dda30864p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x2p+32L : 0x1.791272ee9dd8e28d0c03dda30864p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x2p+32L : 0x1.791272ee9dd8e28d0c03dda30864p+4L : inexact-ok
+= acosh upward ldbl-128 0x2p+32L : 0x1.791272ee9dd8e28d0c03dda30865p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x2p+32L : 0x1.791272ee9dd8e28d0c03dda308p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x2p+32L : 0x1.791272ee9dd8e28d0c03dda3088p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x2p+32L : 0x1.791272ee9dd8e28d0c03dda308p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x2p+32L : 0x1.791272ee9dd8e28d0c03dda3088p+4L : inexact-ok
+acosh 0x1p48
+= acosh downward flt-32 0x1p+48f : 0x2.1f6d68p+4f : inexact-ok
+= acosh tonearest flt-32 0x1p+48f : 0x2.1f6d68p+4f : inexact-ok
+= acosh towardzero flt-32 0x1p+48f : 0x2.1f6d68p+4f : inexact-ok
+= acosh upward flt-32 0x1p+48f : 0x2.1f6d6cp+4f : inexact-ok
+= acosh downward dbl-64 0x1p+48 : 0x2.1f6d6966f28b6p+4 : inexact-ok
+= acosh tonearest dbl-64 0x1p+48 : 0x2.1f6d6966f28b6p+4 : inexact-ok
+= acosh towardzero dbl-64 0x1p+48 : 0x2.1f6d6966f28b6p+4 : inexact-ok
+= acosh upward dbl-64 0x1p+48 : 0x2.1f6d6966f28b8p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+48L : 0x2.1f6d6966f28b649cp+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+48L : 0x2.1f6d6966f28b64ap+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+48L : 0x2.1f6d6966f28b649cp+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+48L : 0x2.1f6d6966f28b64ap+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+48L : 0x2.1f6d6966f28b649cp+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+48L : 0x2.1f6d6966f28b64ap+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+48L : 0x2.1f6d6966f28b649cp+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+48L : 0x2.1f6d6966f28b64ap+4L : inexact-ok
+= acosh downward ldbl-128 0x1p+48L : 0x2.1f6d6966f28b649e1a4956018818p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+48L : 0x2.1f6d6966f28b649e1a4956018818p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+48L : 0x2.1f6d6966f28b649e1a4956018818p+4L : inexact-ok
+= acosh upward ldbl-128 0x1p+48L : 0x2.1f6d6966f28b649e1a495601881ap+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x1p+48L : 0x2.1f6d6966f28b649e1a49560188p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x1p+48L : 0x2.1f6d6966f28b649e1a49560188p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x1p+48L : 0x2.1f6d6966f28b649e1a49560188p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x1p+48L : 0x2.1f6d6966f28b649e1a49560189p+4L : inexact-ok
+acosh 0x1p49
+= acosh downward flt-32 0x2p+48f : 0x2.2a8488p+4f : inexact-ok
+= acosh tonearest flt-32 0x2p+48f : 0x2.2a848cp+4f : inexact-ok
+= acosh towardzero flt-32 0x2p+48f : 0x2.2a8488p+4f : inexact-ok
+= acosh upward flt-32 0x2p+48f : 0x2.2a848cp+4f : inexact-ok
+= acosh downward dbl-64 0x2p+48 : 0x2.2a848ae66fa84p+4 : inexact-ok
+= acosh tonearest dbl-64 0x2p+48 : 0x2.2a848ae66fa86p+4 : inexact-ok
+= acosh towardzero dbl-64 0x2p+48 : 0x2.2a848ae66fa84p+4 : inexact-ok
+= acosh upward dbl-64 0x2p+48 : 0x2.2a848ae66fa86p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x2p+48L : 0x2.2a848ae66fa85c38p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x2p+48L : 0x2.2a848ae66fa85c38p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x2p+48L : 0x2.2a848ae66fa85c38p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x2p+48L : 0x2.2a848ae66fa85c3cp+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x2p+48L : 0x2.2a848ae66fa85c38p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x2p+48L : 0x2.2a848ae66fa85c38p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x2p+48L : 0x2.2a848ae66fa85c38p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x2p+48L : 0x2.2a848ae66fa85c3cp+4L : inexact-ok
+= acosh downward ldbl-128 0x2p+48L : 0x2.2a848ae66fa85c38d6e7913b0b56p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x2p+48L : 0x2.2a848ae66fa85c38d6e7913b0b58p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x2p+48L : 0x2.2a848ae66fa85c38d6e7913b0b56p+4L : inexact-ok
+= acosh upward ldbl-128 0x2p+48L : 0x2.2a848ae66fa85c38d6e7913b0b58p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x2p+48L : 0x2.2a848ae66fa85c38d6e7913b0bp+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x2p+48L : 0x2.2a848ae66fa85c38d6e7913b0bp+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x2p+48L : 0x2.2a848ae66fa85c38d6e7913b0bp+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x2p+48L : 0x2.2a848ae66fa85c38d6e7913b0cp+4L : inexact-ok
+acosh 0x1p50
+= acosh downward flt-32 0x4p+48f : 0x2.359bacp+4f : inexact-ok
+= acosh tonearest flt-32 0x4p+48f : 0x2.359bacp+4f : inexact-ok
+= acosh towardzero flt-32 0x4p+48f : 0x2.359bacp+4f : inexact-ok
+= acosh upward flt-32 0x4p+48f : 0x2.359bbp+4f : inexact-ok
+= acosh downward dbl-64 0x4p+48 : 0x2.359bac65ecc54p+4 : inexact-ok
+= acosh tonearest dbl-64 0x4p+48 : 0x2.359bac65ecc56p+4 : inexact-ok
+= acosh towardzero dbl-64 0x4p+48 : 0x2.359bac65ecc54p+4 : inexact-ok
+= acosh upward dbl-64 0x4p+48 : 0x2.359bac65ecc56p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x4p+48L : 0x2.359bac65ecc553dp+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x4p+48L : 0x2.359bac65ecc553d4p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x4p+48L : 0x2.359bac65ecc553dp+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x4p+48L : 0x2.359bac65ecc553d4p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x4p+48L : 0x2.359bac65ecc553dp+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x4p+48L : 0x2.359bac65ecc553d4p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x4p+48L : 0x2.359bac65ecc553dp+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x4p+48L : 0x2.359bac65ecc553d4p+4L : inexact-ok
+= acosh downward ldbl-128 0x4p+48L : 0x2.359bac65ecc553d39385cc748c56p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x4p+48L : 0x2.359bac65ecc553d39385cc748c56p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x4p+48L : 0x2.359bac65ecc553d39385cc748c56p+4L : inexact-ok
+= acosh upward ldbl-128 0x4p+48L : 0x2.359bac65ecc553d39385cc748c58p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x4p+48L : 0x2.359bac65ecc553d39385cc748cp+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x4p+48L : 0x2.359bac65ecc553d39385cc748cp+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x4p+48L : 0x2.359bac65ecc553d39385cc748cp+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x4p+48L : 0x2.359bac65ecc553d39385cc748dp+4L : inexact-ok
+acosh 0x1p51
+= acosh downward flt-32 0x8p+48f : 0x2.40b2ccp+4f : inexact-ok
+= acosh tonearest flt-32 0x8p+48f : 0x2.40b2ccp+4f : inexact-ok
+= acosh towardzero flt-32 0x8p+48f : 0x2.40b2ccp+4f : inexact-ok
+= acosh upward flt-32 0x8p+48f : 0x2.40b2dp+4f : inexact-ok
+= acosh downward dbl-64 0x8p+48 : 0x2.40b2cde569e24p+4 : inexact-ok
+= acosh tonearest dbl-64 0x8p+48 : 0x2.40b2cde569e24p+4 : inexact-ok
+= acosh towardzero dbl-64 0x8p+48 : 0x2.40b2cde569e24p+4 : inexact-ok
+= acosh upward dbl-64 0x8p+48 : 0x2.40b2cde569e26p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x8p+48L : 0x2.40b2cde569e24b6cp+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x8p+48L : 0x2.40b2cde569e24b7p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x8p+48L : 0x2.40b2cde569e24b6cp+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x8p+48L : 0x2.40b2cde569e24b7p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x8p+48L : 0x2.40b2cde569e24b6cp+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x8p+48L : 0x2.40b2cde569e24b7p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x8p+48L : 0x2.40b2cde569e24b6cp+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x8p+48L : 0x2.40b2cde569e24b7p+4L : inexact-ok
+= acosh downward ldbl-128 0x8p+48L : 0x2.40b2cde569e24b6e502407ae0cc4p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x8p+48L : 0x2.40b2cde569e24b6e502407ae0cc6p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x8p+48L : 0x2.40b2cde569e24b6e502407ae0cc4p+4L : inexact-ok
+= acosh upward ldbl-128 0x8p+48L : 0x2.40b2cde569e24b6e502407ae0cc6p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x8p+48L : 0x2.40b2cde569e24b6e502407ae0cp+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x8p+48L : 0x2.40b2cde569e24b6e502407ae0dp+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x8p+48L : 0x2.40b2cde569e24b6e502407ae0cp+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x8p+48L : 0x2.40b2cde569e24b6e502407ae0dp+4L : inexact-ok
+acosh 0x1p52
+= acosh downward flt-32 0x1p+52f : 0x2.4bc9ecp+4f : inexact-ok
+= acosh tonearest flt-32 0x1p+52f : 0x2.4bc9fp+4f : inexact-ok
+= acosh towardzero flt-32 0x1p+52f : 0x2.4bc9ecp+4f : inexact-ok
+= acosh upward flt-32 0x1p+52f : 0x2.4bc9fp+4f : inexact-ok
+= acosh downward dbl-64 0x1p+52 : 0x2.4bc9ef64e6ff4p+4 : inexact-ok
+= acosh tonearest dbl-64 0x1p+52 : 0x2.4bc9ef64e6ff4p+4 : inexact-ok
+= acosh towardzero dbl-64 0x1p+52 : 0x2.4bc9ef64e6ff4p+4 : inexact-ok
+= acosh upward dbl-64 0x1p+52 : 0x2.4bc9ef64e6ff6p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+52L : 0x2.4bc9ef64e6ff4308p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+52L : 0x2.4bc9ef64e6ff4308p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+52L : 0x2.4bc9ef64e6ff4308p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+52L : 0x2.4bc9ef64e6ff430cp+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+52L : 0x2.4bc9ef64e6ff4308p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+52L : 0x2.4bc9ef64e6ff4308p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+52L : 0x2.4bc9ef64e6ff4308p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+52L : 0x2.4bc9ef64e6ff430cp+4L : inexact-ok
+= acosh downward ldbl-128 0x1p+52L : 0x2.4bc9ef64e6ff43090cc242e78d1p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+52L : 0x2.4bc9ef64e6ff43090cc242e78d1p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+52L : 0x2.4bc9ef64e6ff43090cc242e78d1p+4L : inexact-ok
+= acosh upward ldbl-128 0x1p+52L : 0x2.4bc9ef64e6ff43090cc242e78d12p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x1p+52L : 0x2.4bc9ef64e6ff43090cc242e78dp+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x1p+52L : 0x2.4bc9ef64e6ff43090cc242e78dp+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x1p+52L : 0x2.4bc9ef64e6ff43090cc242e78dp+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x1p+52L : 0x2.4bc9ef64e6ff43090cc242e78ep+4L : inexact-ok
+acosh 0x1p53
+= acosh downward flt-32 0x2p+52f : 0x2.56e11p+4f : inexact-ok
+= acosh tonearest flt-32 0x2p+52f : 0x2.56e11p+4f : inexact-ok
+= acosh towardzero flt-32 0x2p+52f : 0x2.56e11p+4f : inexact-ok
+= acosh upward flt-32 0x2p+52f : 0x2.56e114p+4f : inexact-ok
+= acosh downward dbl-64 0x2p+52 : 0x2.56e110e4641c2p+4 : inexact-ok
+= acosh tonearest dbl-64 0x2p+52 : 0x2.56e110e4641c4p+4 : inexact-ok
+= acosh towardzero dbl-64 0x2p+52 : 0x2.56e110e4641c2p+4 : inexact-ok
+= acosh upward dbl-64 0x2p+52 : 0x2.56e110e4641c4p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x2p+52L : 0x2.56e110e4641c3aap+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x2p+52L : 0x2.56e110e4641c3aa4p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x2p+52L : 0x2.56e110e4641c3aap+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x2p+52L : 0x2.56e110e4641c3aa4p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x2p+52L : 0x2.56e110e4641c3aap+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x2p+52L : 0x2.56e110e4641c3aa4p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x2p+52L : 0x2.56e110e4641c3aap+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x2p+52L : 0x2.56e110e4641c3aa4p+4L : inexact-ok
+= acosh downward ldbl-128 0x2p+52L : 0x2.56e110e4641c3aa3c9607e210d52p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x2p+52L : 0x2.56e110e4641c3aa3c9607e210d54p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x2p+52L : 0x2.56e110e4641c3aa3c9607e210d52p+4L : inexact-ok
+= acosh upward ldbl-128 0x2p+52L : 0x2.56e110e4641c3aa3c9607e210d54p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x2p+52L : 0x2.56e110e4641c3aa3c9607e210dp+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x2p+52L : 0x2.56e110e4641c3aa3c9607e210dp+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x2p+52L : 0x2.56e110e4641c3aa3c9607e210dp+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x2p+52L : 0x2.56e110e4641c3aa3c9607e210ep+4L : inexact-ok
+acosh 0x1p54
+= acosh downward flt-32 0x4p+52f : 0x2.61f83p+4f : inexact-ok
+= acosh tonearest flt-32 0x4p+52f : 0x2.61f834p+4f : inexact-ok
+= acosh towardzero flt-32 0x4p+52f : 0x2.61f83p+4f : inexact-ok
+= acosh upward flt-32 0x4p+52f : 0x2.61f834p+4f : inexact-ok
+= acosh downward dbl-64 0x4p+52 : 0x2.61f83263e1392p+4 : inexact-ok
+= acosh tonearest dbl-64 0x4p+52 : 0x2.61f83263e1394p+4 : inexact-ok
+= acosh towardzero dbl-64 0x4p+52 : 0x2.61f83263e1392p+4 : inexact-ok
+= acosh upward dbl-64 0x4p+52 : 0x2.61f83263e1394p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x4p+52L : 0x2.61f83263e139323cp+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x4p+52L : 0x2.61f83263e139324p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x4p+52L : 0x2.61f83263e139323cp+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x4p+52L : 0x2.61f83263e139324p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x4p+52L : 0x2.61f83263e139323cp+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x4p+52L : 0x2.61f83263e139324p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x4p+52L : 0x2.61f83263e139323cp+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x4p+52L : 0x2.61f83263e139324p+4L : inexact-ok
+= acosh downward ldbl-128 0x4p+52L : 0x2.61f83263e139323e85feb95a8d92p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x4p+52L : 0x2.61f83263e139323e85feb95a8d92p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x4p+52L : 0x2.61f83263e139323e85feb95a8d92p+4L : inexact-ok
+= acosh upward ldbl-128 0x4p+52L : 0x2.61f83263e139323e85feb95a8d94p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x4p+52L : 0x2.61f83263e139323e85feb95a8dp+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x4p+52L : 0x2.61f83263e139323e85feb95a8ep+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x4p+52L : 0x2.61f83263e139323e85feb95a8dp+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x4p+52L : 0x2.61f83263e139323e85feb95a8ep+4L : inexact-ok
+acosh 0x1p55
+= acosh downward flt-32 0x8p+52f : 0x2.6d0f5p+4f : inexact-ok
+= acosh tonearest flt-32 0x8p+52f : 0x2.6d0f54p+4f : inexact-ok
+= acosh towardzero flt-32 0x8p+52f : 0x2.6d0f5p+4f : inexact-ok
+= acosh upward flt-32 0x8p+52f : 0x2.6d0f54p+4f : inexact-ok
+= acosh downward dbl-64 0x8p+52 : 0x2.6d0f53e35e562p+4 : inexact-ok
+= acosh tonearest dbl-64 0x8p+52 : 0x2.6d0f53e35e562p+4 : inexact-ok
+= acosh towardzero dbl-64 0x8p+52 : 0x2.6d0f53e35e562p+4 : inexact-ok
+= acosh upward dbl-64 0x8p+52 : 0x2.6d0f53e35e564p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x8p+52L : 0x2.6d0f53e35e5629d8p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x8p+52L : 0x2.6d0f53e35e5629d8p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x8p+52L : 0x2.6d0f53e35e5629d8p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x8p+52L : 0x2.6d0f53e35e5629dcp+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x8p+52L : 0x2.6d0f53e35e5629d8p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x8p+52L : 0x2.6d0f53e35e5629d8p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x8p+52L : 0x2.6d0f53e35e5629d8p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x8p+52L : 0x2.6d0f53e35e5629dcp+4L : inexact-ok
+= acosh downward ldbl-128 0x8p+52L : 0x2.6d0f53e35e5629d9429cf4940dd2p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x8p+52L : 0x2.6d0f53e35e5629d9429cf4940dd2p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x8p+52L : 0x2.6d0f53e35e5629d9429cf4940dd2p+4L : inexact-ok
+= acosh upward ldbl-128 0x8p+52L : 0x2.6d0f53e35e5629d9429cf4940dd4p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x8p+52L : 0x2.6d0f53e35e5629d9429cf4940dp+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x8p+52L : 0x2.6d0f53e35e5629d9429cf4940ep+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x8p+52L : 0x2.6d0f53e35e5629d9429cf4940dp+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x8p+52L : 0x2.6d0f53e35e5629d9429cf4940ep+4L : inexact-ok
+acosh 0x1p56
+= acosh downward flt-32 0x1p+56f : 0x2.782674p+4f : inexact-ok
+= acosh tonearest flt-32 0x1p+56f : 0x2.782674p+4f : inexact-ok
+= acosh towardzero flt-32 0x1p+56f : 0x2.782674p+4f : inexact-ok
+= acosh upward flt-32 0x1p+56f : 0x2.782678p+4f : inexact-ok
+= acosh downward dbl-64 0x1p+56 : 0x2.78267562db732p+4 : inexact-ok
+= acosh tonearest dbl-64 0x1p+56 : 0x2.78267562db732p+4 : inexact-ok
+= acosh towardzero dbl-64 0x1p+56 : 0x2.78267562db732p+4 : inexact-ok
+= acosh upward dbl-64 0x1p+56 : 0x2.78267562db734p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+56L : 0x2.78267562db73217p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+56L : 0x2.78267562db732174p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+56L : 0x2.78267562db73217p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+56L : 0x2.78267562db732174p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+56L : 0x2.78267562db73217p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+56L : 0x2.78267562db732174p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+56L : 0x2.78267562db73217p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+56L : 0x2.78267562db732174p+4L : inexact-ok
+= acosh downward ldbl-128 0x1p+56L : 0x2.78267562db732173ff3b2fcd8e1p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+56L : 0x2.78267562db732173ff3b2fcd8e12p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+56L : 0x2.78267562db732173ff3b2fcd8e1p+4L : inexact-ok
+= acosh upward ldbl-128 0x1p+56L : 0x2.78267562db732173ff3b2fcd8e12p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x1p+56L : 0x2.78267562db732173ff3b2fcd8ep+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x1p+56L : 0x2.78267562db732173ff3b2fcd8ep+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x1p+56L : 0x2.78267562db732173ff3b2fcd8ep+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x1p+56L : 0x2.78267562db732173ff3b2fcd8fp+4L : inexact-ok
+acosh 0x1p57
+= acosh downward flt-32 0x2p+56f : 0x2.833d94p+4f : inexact-ok
+= acosh tonearest flt-32 0x2p+56f : 0x2.833d98p+4f : inexact-ok
+= acosh towardzero flt-32 0x2p+56f : 0x2.833d94p+4f : inexact-ok
+= acosh upward flt-32 0x2p+56f : 0x2.833d98p+4f : inexact-ok
+= acosh downward dbl-64 0x2p+56 : 0x2.833d96e2589p+4 : inexact-ok
+= acosh tonearest dbl-64 0x2p+56 : 0x2.833d96e258902p+4 : inexact-ok
+= acosh towardzero dbl-64 0x2p+56 : 0x2.833d96e2589p+4 : inexact-ok
+= acosh upward dbl-64 0x2p+56 : 0x2.833d96e258902p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x2p+56L : 0x2.833d96e25890190cp+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x2p+56L : 0x2.833d96e25890191p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x2p+56L : 0x2.833d96e25890190cp+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x2p+56L : 0x2.833d96e25890191p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x2p+56L : 0x2.833d96e25890190cp+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x2p+56L : 0x2.833d96e25890191p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x2p+56L : 0x2.833d96e25890190cp+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x2p+56L : 0x2.833d96e25890191p+4L : inexact-ok
+= acosh downward ldbl-128 0x2p+56L : 0x2.833d96e25890190ebbd96b070e5p+4L : inexact-ok
+= acosh tonearest ldbl-128 0x2p+56L : 0x2.833d96e25890190ebbd96b070e5p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x2p+56L : 0x2.833d96e25890190ebbd96b070e5p+4L : inexact-ok
+= acosh upward ldbl-128 0x2p+56L : 0x2.833d96e25890190ebbd96b070e52p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x2p+56L : 0x2.833d96e25890190ebbd96b070ep+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x2p+56L : 0x2.833d96e25890190ebbd96b070ep+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x2p+56L : 0x2.833d96e25890190ebbd96b070ep+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x2p+56L : 0x2.833d96e25890190ebbd96b070fp+4L : inexact-ok
+acosh 0x1p58
+= acosh downward flt-32 0x4p+56f : 0x2.8e54b8p+4f : inexact-ok
+= acosh tonearest flt-32 0x4p+56f : 0x2.8e54b8p+4f : inexact-ok
+= acosh towardzero flt-32 0x4p+56f : 0x2.8e54b8p+4f : inexact-ok
+= acosh upward flt-32 0x4p+56f : 0x2.8e54bcp+4f : inexact-ok
+= acosh downward dbl-64 0x4p+56 : 0x2.8e54b861d5adp+4 : inexact-ok
+= acosh tonearest dbl-64 0x4p+56 : 0x2.8e54b861d5ad2p+4 : inexact-ok
+= acosh towardzero dbl-64 0x4p+56 : 0x2.8e54b861d5adp+4 : inexact-ok
+= acosh upward dbl-64 0x4p+56 : 0x2.8e54b861d5ad2p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x4p+56L : 0x2.8e54b861d5ad10a8p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x4p+56L : 0x2.8e54b861d5ad10a8p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x4p+56L : 0x2.8e54b861d5ad10a8p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x4p+56L : 0x2.8e54b861d5ad10acp+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x4p+56L : 0x2.8e54b861d5ad10a8p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x4p+56L : 0x2.8e54b861d5ad10a8p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x4p+56L : 0x2.8e54b861d5ad10a8p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x4p+56L : 0x2.8e54b861d5ad10acp+4L : inexact-ok
+= acosh downward ldbl-128 0x4p+56L : 0x2.8e54b861d5ad10a97877a6408e8ep+4L : inexact-ok
+= acosh tonearest ldbl-128 0x4p+56L : 0x2.8e54b861d5ad10a97877a6408e9p+4L : inexact-ok
+= acosh towardzero ldbl-128 0x4p+56L : 0x2.8e54b861d5ad10a97877a6408e8ep+4L : inexact-ok
+= acosh upward ldbl-128 0x4p+56L : 0x2.8e54b861d5ad10a97877a6408e9p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x4p+56L : 0x2.8e54b861d5ad10a97877a6408ep+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x4p+56L : 0x2.8e54b861d5ad10a97877a6408fp+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x4p+56L : 0x2.8e54b861d5ad10a97877a6408ep+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x4p+56L : 0x2.8e54b861d5ad10a97877a6408fp+4L : inexact-ok
+acosh 0x1p59
+= acosh downward flt-32 0x8p+56f : 0x2.996bd8p+4f : inexact-ok
+= acosh tonearest flt-32 0x8p+56f : 0x2.996bd8p+4f : inexact-ok
+= acosh towardzero flt-32 0x8p+56f : 0x2.996bd8p+4f : inexact-ok
+= acosh upward flt-32 0x8p+56f : 0x2.996bdcp+4f : inexact-ok
+= acosh downward dbl-64 0x8p+56 : 0x2.996bd9e152cap+4 : inexact-ok
+= acosh tonearest dbl-64 0x8p+56 : 0x2.996bd9e152cap+4 : inexact-ok
+= acosh towardzero dbl-64 0x8p+56 : 0x2.996bd9e152cap+4 : inexact-ok
+= acosh upward dbl-64 0x8p+56 : 0x2.996bd9e152ca2p+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x8p+56L : 0x2.996bd9e152ca0844p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x8p+56L : 0x2.996bd9e152ca0844p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x8p+56L : 0x2.996bd9e152ca0844p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x8p+56L : 0x2.996bd9e152ca0848p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x8p+56L : 0x2.996bd9e152ca0844p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x8p+56L : 0x2.996bd9e152ca0844p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x8p+56L : 0x2.996bd9e152ca0844p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x8p+56L : 0x2.996bd9e152ca0848p+4L : inexact-ok
+= acosh downward ldbl-128 0x8p+56L : 0x2.996bd9e152ca08443515e17a0ecep+4L : inexact-ok
+= acosh tonearest ldbl-128 0x8p+56L : 0x2.996bd9e152ca08443515e17a0edp+4L : inexact-ok
+= acosh towardzero ldbl-128 0x8p+56L : 0x2.996bd9e152ca08443515e17a0ecep+4L : inexact-ok
+= acosh upward ldbl-128 0x8p+56L : 0x2.996bd9e152ca08443515e17a0edp+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x8p+56L : 0x2.996bd9e152ca08443515e17a0ep+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x8p+56L : 0x2.996bd9e152ca08443515e17a0fp+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x8p+56L : 0x2.996bd9e152ca08443515e17a0ep+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x8p+56L : 0x2.996bd9e152ca08443515e17a0fp+4L : inexact-ok
+acosh 0x1p100
+= acosh downward flt-32 0x1p+100f : 0x4.60203p+4f : inexact-ok
+= acosh tonearest flt-32 0x1p+100f : 0x4.602038p+4f : inexact-ok
+= acosh towardzero flt-32 0x1p+100f : 0x4.60203p+4f : inexact-ok
+= acosh upward flt-32 0x1p+100f : 0x4.602038p+4f : inexact-ok
+= acosh downward dbl-64 0x1p+100 : 0x4.6020374c5c6d8p+4 : inexact-ok
+= acosh tonearest dbl-64 0x1p+100 : 0x4.6020374c5c6dcp+4 : inexact-ok
+= acosh towardzero dbl-64 0x1p+100 : 0x4.6020374c5c6d8p+4 : inexact-ok
+= acosh upward dbl-64 0x1p+100 : 0x4.6020374c5c6dcp+4 : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+100L : 0x4.6020374c5c6db008p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+100L : 0x4.6020374c5c6db01p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+100L : 0x4.6020374c5c6db008p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+100L : 0x4.6020374c5c6db01p+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+100L : 0x4.6020374c5c6db008p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+100L : 0x4.6020374c5c6db01p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+100L : 0x4.6020374c5c6db008p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+100L : 0x4.6020374c5c6db01p+4L : inexact-ok
+= acosh downward ldbl-128 0x1p+100L : 0x4.6020374c5c6db00c6a6d5daf98ecp+4L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+100L : 0x4.6020374c5c6db00c6a6d5daf98ecp+4L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+100L : 0x4.6020374c5c6db00c6a6d5daf98ecp+4L : inexact-ok
+= acosh upward ldbl-128 0x1p+100L : 0x4.6020374c5c6db00c6a6d5daf98fp+4L : inexact-ok
+= acosh downward ldbl-128ibm 0x1p+100L : 0x4.6020374c5c6db00c6a6d5daf98p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x1p+100L : 0x4.6020374c5c6db00c6a6d5daf98p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x1p+100L : 0x4.6020374c5c6db00c6a6d5daf98p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0x1p+100L : 0x4.6020374c5c6db00c6a6d5daf9ap+4L : inexact-ok
+acosh 0x1p500
+= acosh downward flt-32 0xf.fffffp+124f : 0x5.96a7ep+4f : inexact-ok
+= acosh tonearest flt-32 0xf.fffffp+124f : 0x5.96a7ep+4f : inexact-ok
+= acosh towardzero flt-32 0xf.fffffp+124f : 0x5.96a7ep+4f : inexact-ok
+= acosh upward flt-32 0xf.fffffp+124f : 0x5.96a7e8p+4f : inexact-ok
+= acosh downward dbl-64 0xf.fffffp+124 : 0x5.96a7e12e0b988p+4 : inexact-ok
+= acosh tonearest dbl-64 0xf.fffffp+124 : 0x5.96a7e12e0b98cp+4 : inexact-ok
+= acosh towardzero dbl-64 0xf.fffffp+124 : 0x5.96a7e12e0b988p+4 : inexact-ok
+= acosh upward dbl-64 0xf.fffffp+124 : 0x5.96a7e12e0b98cp+4 : inexact-ok
+= acosh downward ldbl-96-intel 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0xf.fffffp+124L : 0x5.96a7e12e0b98bdp+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0xf.fffffp+124L : 0x5.96a7e12e0b98bdp+4L : inexact-ok
+= acosh downward ldbl-128 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a4468p+4L : inexact-ok
+= acosh tonearest ldbl-128 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a4468p+4L : inexact-ok
+= acosh towardzero ldbl-128 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a4468p+4L : inexact-ok
+= acosh upward ldbl-128 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a44684p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a446p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a446p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a446p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a448p+4L : inexact-ok
+= acosh downward dbl-64 0x1p+500 : 0x1.5b4448e7fd9bp+8 : inexact-ok
+= acosh tonearest dbl-64 0x1p+500 : 0x1.5b4448e7fd9b1p+8 : inexact-ok
+= acosh towardzero dbl-64 0x1p+500 : 0x1.5b4448e7fd9bp+8 : inexact-ok
+= acosh upward dbl-64 0x1p+500 : 0x1.5b4448e7fd9b1p+8 : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+500L : 0x1.5b4448e7fd9b091cp+8L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+500L : 0x1.5b4448e7fd9b091ep+8L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+500L : 0x1.5b4448e7fd9b091cp+8L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+500L : 0x1.5b4448e7fd9b091ep+8L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+500L : 0x1.5b4448e7fd9b091cp+8L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+500L : 0x1.5b4448e7fd9b091ep+8L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+500L : 0x1.5b4448e7fd9b091cp+8L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+500L : 0x1.5b4448e7fd9b091ep+8L : inexact-ok
+= acosh downward ldbl-128 0x1p+500L : 0x1.5b4448e7fd9b091d321a9e787fbap+8L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+500L : 0x1.5b4448e7fd9b091d321a9e787fbap+8L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+500L : 0x1.5b4448e7fd9b091d321a9e787fbap+8L : inexact-ok
+= acosh upward ldbl-128 0x1p+500L : 0x1.5b4448e7fd9b091d321a9e787fbbp+8L : inexact-ok
+= acosh downward ldbl-128ibm 0x1p+500L : 0x1.5b4448e7fd9b091d321a9e787f8p+8L : inexact-ok
+= acosh tonearest ldbl-128ibm 0x1p+500L : 0x1.5b4448e7fd9b091d321a9e787f8p+8L : inexact-ok
+= acosh towardzero ldbl-128ibm 0x1p+500L : 0x1.5b4448e7fd9b091d321a9e787f8p+8L : inexact-ok
+= acosh upward ldbl-128ibm 0x1p+500L : 0x1.5b4448e7fd9b091d321a9e788p+8L : inexact-ok
+acosh 0x1p5000
+= acosh downward flt-32 0xf.fffffp+124f : 0x5.96a7ep+4f : inexact-ok
+= acosh tonearest flt-32 0xf.fffffp+124f : 0x5.96a7ep+4f : inexact-ok
+= acosh towardzero flt-32 0xf.fffffp+124f : 0x5.96a7ep+4f : inexact-ok
+= acosh upward flt-32 0xf.fffffp+124f : 0x5.96a7e8p+4f : inexact-ok
+= acosh downward dbl-64 0xf.fffffp+124 : 0x5.96a7e12e0b988p+4 : inexact-ok
+= acosh tonearest dbl-64 0xf.fffffp+124 : 0x5.96a7e12e0b98cp+4 : inexact-ok
+= acosh towardzero dbl-64 0xf.fffffp+124 : 0x5.96a7e12e0b988p+4 : inexact-ok
+= acosh upward dbl-64 0xf.fffffp+124 : 0x5.96a7e12e0b98cp+4 : inexact-ok
+= acosh downward ldbl-96-intel 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh tonearest ldbl-96-intel 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh towardzero ldbl-96-intel 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh upward ldbl-96-intel 0xf.fffffp+124L : 0x5.96a7e12e0b98bdp+4L : inexact-ok
+= acosh downward ldbl-96-m68k 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf8p+4L : inexact-ok
+= acosh upward ldbl-96-m68k 0xf.fffffp+124L : 0x5.96a7e12e0b98bdp+4L : inexact-ok
+= acosh downward ldbl-128 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a4468p+4L : inexact-ok
+= acosh tonearest ldbl-128 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a4468p+4L : inexact-ok
+= acosh towardzero ldbl-128 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a4468p+4L : inexact-ok
+= acosh upward ldbl-128 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a44684p+4L : inexact-ok
+= acosh downward ldbl-128ibm 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a446p+4L : inexact-ok
+= acosh tonearest ldbl-128ibm 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a446p+4L : inexact-ok
+= acosh towardzero ldbl-128ibm 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a446p+4L : inexact-ok
+= acosh upward ldbl-128ibm 0xf.fffffp+124L : 0x5.96a7e12e0b98bcf90bb682a448p+4L : inexact-ok
+= acosh downward dbl-64 0xf.ffffffffffff8p+1020 : 0x2.c679d1f73f0fap+8 : inexact-ok
+= acosh tonearest dbl-64 0xf.ffffffffffff8p+1020 : 0x2.c679d1f73f0fcp+8 : inexact-ok
+= acosh towardzero dbl-64 0xf.ffffffffffff8p+1020 : 0x2.c679d1f73f0fap+8 : inexact-ok
+= acosh upward dbl-64 0xf.ffffffffffff8p+1020 : 0x2.c679d1f73f0fcp+8 : inexact-ok
+= acosh downward ldbl-96-intel 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb62p+8L : inexact-ok
+= acosh tonearest ldbl-96-intel 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb62p+8L : inexact-ok
+= acosh towardzero ldbl-96-intel 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb62p+8L : inexact-ok
+= acosh upward ldbl-96-intel 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb624p+8L : inexact-ok
+= acosh downward ldbl-96-m68k 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb62p+8L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb62p+8L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb62p+8L : inexact-ok
+= acosh upward ldbl-96-m68k 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb624p+8L : inexact-ok
+= acosh downward ldbl-128 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb620d358b213a7cep+8L : inexact-ok
+= acosh tonearest ldbl-128 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb620d358b213a7dp+8L : inexact-ok
+= acosh towardzero ldbl-128 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb620d358b213a7cep+8L : inexact-ok
+= acosh upward ldbl-128 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb620d358b213a7dp+8L : inexact-ok
+= acosh downward ldbl-128ibm 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb620d358b213a7p+8L : inexact-ok
+= acosh tonearest ldbl-128ibm 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb620d358b213a8p+8L : inexact-ok
+= acosh towardzero ldbl-128ibm 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb620d358b213a7p+8L : inexact-ok
+= acosh upward ldbl-128ibm 0xf.ffffffffffff8p+1020L : 0x2.c679d1f73f0fb620d358b213a8p+8L : inexact-ok
+= acosh downward ldbl-96-intel 0x1p+5000L : 0xd.8a6dd63831ae0fdp+8L : inexact-ok
+= acosh tonearest ldbl-96-intel 0x1p+5000L : 0xd.8a6dd63831ae0fep+8L : inexact-ok
+= acosh towardzero ldbl-96-intel 0x1p+5000L : 0xd.8a6dd63831ae0fdp+8L : inexact-ok
+= acosh upward ldbl-96-intel 0x1p+5000L : 0xd.8a6dd63831ae0fep+8L : inexact-ok
+= acosh downward ldbl-96-m68k 0x1p+5000L : 0xd.8a6dd63831ae0fdp+8L : inexact-ok
+= acosh tonearest ldbl-96-m68k 0x1p+5000L : 0xd.8a6dd63831ae0fep+8L : inexact-ok
+= acosh towardzero ldbl-96-m68k 0x1p+5000L : 0xd.8a6dd63831ae0fdp+8L : inexact-ok
+= acosh upward ldbl-96-m68k 0x1p+5000L : 0xd.8a6dd63831ae0fep+8L : inexact-ok
+= acosh downward ldbl-128 0x1p+5000L : 0xd.8a6dd63831ae0fdceaf12f64a52p+8L : inexact-ok
+= acosh tonearest ldbl-128 0x1p+5000L : 0xd.8a6dd63831ae0fdceaf12f64a528p+8L : inexact-ok
+= acosh towardzero ldbl-128 0x1p+5000L : 0xd.8a6dd63831ae0fdceaf12f64a52p+8L : inexact-ok
+= acosh upward ldbl-128 0x1p+5000L : 0xd.8a6dd63831ae0fdceaf12f64a528p+8L : inexact-ok
+= acosh downward ldbl-128 0xf.ffffffffffffbffffffffffffcp+1020L : 0x2.c679d1f73f0fb624d358b213a7cep+8L : inexact-ok
+= acosh tonearest ldbl-128 0xf.ffffffffffffbffffffffffffcp+1020L : 0x2.c679d1f73f0fb624d358b213a7dp+8L : inexact-ok
+= acosh towardzero ldbl-128 0xf.ffffffffffffbffffffffffffcp+1020L : 0x2.c679d1f73f0fb624d358b213a7cep+8L : inexact-ok
+= acosh upward ldbl-128 0xf.ffffffffffffbffffffffffffcp+1020L : 0x2.c679d1f73f0fb624d358b213a7dp+8L : inexact-ok
+= acosh downward ldbl-128ibm 0xf.ffffffffffffbffffffffffffcp+1020L : 0x2.c679d1f73f0fb624d358b213a7p+8L : inexact-ok
+= acosh tonearest ldbl-128ibm 0xf.ffffffffffffbffffffffffffcp+1020L : 0x2.c679d1f73f0fb624d358b213a8p+8L : inexact-ok
+= acosh towardzero ldbl-128ibm 0xf.ffffffffffffbffffffffffffcp+1020L : 0x2.c679d1f73f0fb624d358b213a7p+8L : inexact-ok
+= acosh upward ldbl-128ibm 0xf.ffffffffffffbffffffffffffcp+1020L : 0x2.c679d1f73f0fb624d358b213a8p+8L : inexact-ok
 acosh max no-test-inline
 = acosh downward flt-32 0xf.fffffp+124f : 0x5.96a7ep+4f : no-test-inline inexact-ok
 = acosh tonearest flt-32 0xf.fffffp+124f : 0x5.96a7ep+4f : no-test-inline inexact-ok
diff --git a/sysdeps/i386/fpu/libm-test-ulps b/sysdeps/i386/fpu/libm-test-ulps
index 5e79307..0977557 100644
--- a/sysdeps/i386/fpu/libm-test-ulps
+++ b/sysdeps/i386/fpu/libm-test-ulps
@@ -21,6 +21,10 @@ Function: "acos_upward":
 ildouble: 1
 ldouble: 1
 
+Function: "acosh":
+ildouble: 1
+ldouble: 1
+
 Function: "acosh_downward":
 ildouble: 2
 ldouble: 2
diff --git a/sysdeps/ieee754/ldbl-128ibm/e_acoshl.c b/sysdeps/ieee754/ldbl-128ibm/e_acoshl.c
index b0b33f7..cab1da9 100644
--- a/sysdeps/ieee754/ldbl-128ibm/e_acoshl.c
+++ b/sysdeps/ieee754/ldbl-128ibm/e_acoshl.c
@@ -44,14 +44,14 @@ __ieee754_acoshl(long double x)
 	EXTRACT_WORDS64 (lx, xlo);
 	if(hx<0x3ff0000000000000LL) {		/* x < 1 */
 	    return (x-x)/(x-x);
-	} else if(hx >=0x41b0000000000000LL) {	/* x > 2**28 */
+	} else if(hx >=0x4370000000000000LL) {	/* x >= 2**56 */
 	    if(hx >=0x7ff0000000000000LL) {	/* x is inf of NaN */
 		return x+x;
 	    } else
 		return __ieee754_logl(x)+ln2;	/* acosh(huge)=log(2x) */
 	} else if (((hx-0x3ff0000000000000LL)|(lx&0x7fffffffffffffffLL))==0) {
 	    return 0.0;			/* acosh(1) = 0 */
-	} else if (hx > 0x4000000000000000LL) {	/* 2**28 > x > 2 */
+	} else if (hx > 0x4000000000000000LL) {	/* 2**56 > x > 2 */
 	    t=x*x;
 	    return __ieee754_logl(2.0*x-one/(x+__ieee754_sqrtl(t-one)));
 	} else {			/* 1<x<2 */
diff --git a/sysdeps/x86_64/fpu/libm-test-ulps b/sysdeps/x86_64/fpu/libm-test-ulps
index 36e1b76..cd9e44f 100644
--- a/sysdeps/x86_64/fpu/libm-test-ulps
+++ b/sysdeps/x86_64/fpu/libm-test-ulps
@@ -29,20 +29,33 @@ ldouble: 1
 
 Function: "acosh":
 double: 1
+float: 1
 idouble: 1
+ifloat: 1
+ildouble: 1
+ldouble: 1
 
 Function: "acosh_downward":
+double: 1
 float: 1
-ildouble: 1
+idouble: 1
+ifloat: 1
+ildouble: 2
 ldouble: 2
 
 Function: "acosh_towardzero":
+double: 1
 float: 1
-ildouble: 1
+idouble: 1
+ifloat: 1
+ildouble: 2
 ldouble: 2
 
 Function: "acosh_upward":
 double: 1
+float: 1
+idouble: 1
+ifloat: 1
 ildouble: 1
 ldouble: 1
 
-- 
2.3.5

